example/linux/Makefile

```diff
@@ -70,12 +69,10 @@ SOURCES+=$(WRAPPER_SOURCES)
 
 # Headers
 WRAPPER_INCLUDE_DIR=$(WRAPPER_ROOT)/include
-INCLUDE_DIRS=$(FLUTTER_LIBRARY_COPY_DIR) $(WRAPPER_INCLUDE_DIR)
+INCLUDE_DIRS=$(FLUTTER_APP_CACHE_DIR) $(WRAPPER_INCLUDE_DIR)
 
 # The stamp file created by copy_flutter_files.
-FLUTTER_COPY_STAMP_FILE=$(FLUTTER_LIBRARY_COPY_DIR)/.last_copied_flutter_version
-# The Flutter engine version file.
-FLUTTER_ENGINE_VERSION_FILE=$(FLUTTER_ROOT)/bin/internal/engine.version
+FLUTTER_COPY_STAMP_FILE=$(FLUTTER_APP_CACHE_DIR)/generated_flutter_root
```

Comment:
This should just be removed, along with the dependency below using it; it no longer makes sense to have the Makefile have a concept of when the copies need to be updated since it doesn't have a way to copy them.

Although now that I'm thinking about it that way, there's a structural problem with this pair of patches: as written, `make` will no longer work reliably, because the copies of the files won't be updated. We need to follow the macOS model where there's a script that we can call to do copying and such but that doesn't invoke `make`, so that `make` can call it. Sorry I missed this when reviewing the other patch.

Reply:
The tool re-entrance requirements add a lot of complexity to this process - It only kind of works at best too.

It is a requirement for iOS due to Xcode and for Android due to Gradle, but I don't see a compelling case yet for Windows or Linux.

Reply:
> It is a requirement for iOS due to Xcode and for Android due to Gradle, but I don't see a compelling case yet for Windows or Linux.

I'm not sure I follow. How is Xcode fundamentally different from Visual Studio, or Make (or whatever other "native" build system we settle on for Linux)? In all these cases, the use case seems to be the same: "I want to build with a native workflow I'm used to for my platform"

Reply:
These workflows do not work if you don't run flutter first - making the tooling call to copy the assets over re-entrant does not change this

Reply:
Only because we don't have a `flutter create` pre-populating some of the generated files, right?

There's a difference between needing to have run the flutter tool once in the past in order to do a build with the underlying build system, and not being able to do a build with the underlying build system in the future either.

And I'm still not clear on how Xcode is different here.

Reply:
What is the workflow we are designing for here? Is this for changing local engine sources, or for working with the native application code?

Reply:
What I mean is, lets go back about 6 steps:

1. The makefile copies the required sources into its own cache directory.
2. The tool can write a config flag which overwrites the default for local engine builds.
3. ???
4 Profit

Reply:
I don't want to design a build system for one workflow though; we should have a build system that supports multiple workflows. Primary cases:
- Just being able to `flutter run`
- Using local engine builds
- Working on the native application code (for which being able to use the underlying native build directly is likely to be easiest)

The iOS/macOS structure seems to strike a good balance of supporting all of that. I'd like the Windows and Linux builds to do balance them as well, so using that structure (where the flutter tool sets configuration and invokes the native build, and the native build has access to a script that copies necessary resources) seems like a good option.

Reply:
In this case though the native build either needs to know the directory location to copy, or it needs to know the location of a script to invoke that can do so for it. This is where I'll plead ignorance a bit in that I don't see too much of a difference between the two (that said, i'll defer to your judgement here).

So I will create an asset sync script for Linux (and Windows) which can be invoked for either a bin/cache or local engine configuration. This will copy the sources into linux/flutter and windows/flutter. flutter build will be responsible for setting a generated_flutter_root file or a build mode override. 

I think that should satisfy the requirement that the native build is still directly usable, while still delegating the heavy lifting to code controlled by the flutter tool

Reply:
> I don't see too much of a difference between the two

The main difference I see is that it gives us the ability to put a lot of logic in the tool itself, and change that logic over time without breaking things:
- Details of where the caches live and how they are updated
- Details of how we decide whether re-copying is necessary (for eventually fixing the alway-relink regression we've introduced to FDE, for instance)
- Understanding of how local build overrides work

> So I will create an asset sync script for Linux (and Windows) which can be invoked for either a bin/cache or local engine configuration.

SGTM. We should be able to use environment variables to communicate local build overrides, as in the macOS script. For Windows, generated properties files can set values that will be in the build environment, and for Linux we can either do another file read in the Makefile like for the Flutter root (eventually I think we'll need a build system with an actual property management system that can replace using files like that), or just for now make the setting non-persistent on Linux and set in in the environment when invoking `make` from the tool.